tap_chargebee/streams/base.py

```diff
@@ -243,10 +243,14 @@ def sync_data(self):
             if self.ENTITY == 'coupon':
                 for coupon in Util.coupons:
                     to_write.append(coupon)
-            if self.ENTITY == 'transaction':
-                # store ids to clean dupplicates
-                to_write = [record for record in to_write if record["id"] not in ids]
-                ids.extend([trans["id"] for trans in to_write])
+            if self.ENTITY in ['transaction', 'subscription']:
+                # store ids to clean dupplicates, keep only the last appearance of a record id
+                new_records = []
+                for record in reversed(to_write):
```

Comment:
if the goal is to keep the latest updated record, I think this will only work if the duplicates are in the same response page, if that's not that relevant please ignore.

Reply:
The current behavior is duplicates with the exact same values besides `cancel_reason_code`, which means that they would come in the same page. To guarantee that we don't have duplicates across response pages I would need to collect all records and write it at the end, which would be more disruptive and not that aligned with the current behavior.

Reply:
What do you think @hsyyid ?

Reply:
Pushed a fix for that